test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -396,6 +396,128 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 
 				deleteVolumeSnapshot(ctx, f, dc, sr, pattern, vscontent)
 			})
+
+			ginkgo.It("should verify that deleting snapshot does not delete the source volume (persistent)", func(ctx context.Context) {
+				if pattern.VolType == storageframework.GenericEphemeralVolume {
+					e2eskipper.Skipf("volume type %q is ephemeral", pattern.VolType)
+				}
+				init(ctx)
+
+				pvc = volumeResource.Pvc
+				sc = volumeResource.Sc
+
+				// The pod should be in the Success state.
+				ginkgo.By("[init] check pod success")
+				pod, err = cs.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "Failed to fetch pod: %v", err)
+				framework.ExpectNoError(e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, cs, pod.Name, pod.Namespace, f.Timeouts.PodStartSlow))
+				// Sync the pod to know additional fields.
+				pod, err = cs.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "Failed to fetch pod: %v", err)
+
+				ginkgo.By("[init] checking the claim")
+				err = e2epv.WaitForPersistentVolumeClaimPhase(ctx, v1.ClaimBound, cs, pvc.Namespace, pvc.Name, framework.Poll, f.Timeouts.ClaimProvision)
+				framework.ExpectNoError(err)
+				// Get new copy of the claim.
+				pvc, err = cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				// Get the bound PV.
+				ginkgo.By("[init] checking the PV")
+				pv, err := cs.CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				// Store the original PV name for later verification
+				originalPVName := pv.Name
+
+				// Delete the pod to force NodeUnpublishVolume.
+				ginkgo.By("[init] deleting the pod")
+				StopPod(ctx, cs, pod)
+
+				// Wait for volume to be fully detached from the node
```

Comment:
The comment better to move before line:445, it seems a bit mislead here.

Reply:
Updated, thanks for the comments.